### PR TITLE
first commit

### PR DESCRIPTION
**Pull request by ygotthilf**

### CODE REVIEW

**Comment by ltbesh-cp** [comment]
What an awesome project !

File: src/middlewares/childLogger.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+// // //
+
+/**
+ * Append a logger to the request. The logger is created with `requestId: req.requestId`
+ * @see middleware requestId.js
+ * @param  {Object} logger The logger (must have a .child method)
+ * @return {Function}        the middleware
+ */
+function setup(logger) {
+  assert.ok(logger, 'Logger is missing');
```
Comment (jean-k-cp) [inline src/middlewares/childLogger.js]: you can also assert here that it respects the contract:  (must have a .child method)

File: src/middlewares/requestId.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+'use strict';
+
+const uuid = require('node-uuid');
+
+module.exports = config;
```
Comment (jean-k-cp) [inline src/middlewares/requestId.js]: config is a little confusing name, it is usually used for the config object...

File: src/middlewares/httpAccessLogger.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+ * @param  {Object} opts.morgan See Morgan options
+ * @param  {Function(req)} opts.info  A function which return a method to log a message (by default Bunyan `req.logger` logger is used)
+ * @param  {Function(req)} opts.isDebug  Allow to display more information whether true (by default Bunyan `req.logger` logger is used)
+ * @param  {Object} opts.tokens  Additionnal Morgan tokens
+ * @param  {Function(req, res)} opts.tokens.authorization This token displays the request jwt token by using `req.token`
+ * @param  {Function(req, res)} opts.tokens.request-id  This token displays the request unique id : `req.requestId`
+ * @see childLogger.js
+ * @see jwtToken.js
+ * @see requestId.js
+ */
```
Comment (melfardi) [inline src/middlewares/httpAccessLogger.js]: JSDoc without body ?

File: src/middlewares/httpAccessLogger.js [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+  * Morgan is a good tools to fetch request data (like duration, http version, ...), but it only streams string instead of Json.
+  * Then we don't use morgan stream and override morgan formatter to log Json request data.
+  *
+ * @param  {Object} opts Options
+ * @return {Function}      the middleware
+ */
+function setup(opts) {
+  const options = _.defaults({
+    morgan: {
+      stream: { write: () => { /* DO NOTHING */ } }
```
Comment (melfardi) [inline src/middlewares/httpAccessLogger.js]: You can also use `_.noop`

File: src/middlewares/httpAccessLogger.js [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    },
+    log: (req) => req.logger.info.bind(req.logger),
+    isDebug: (req) => req.logger.level() < bunyan.INFO,
```
Comment (melfardi) [inline src/middlewares/httpAccessLogger.js]: You can omit parenthesis on single parameter arrow functions.

File: src/middlewares/jwtToken.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+   * Middleware
+   * @param  {Object}   req  Express request
+   * @param  {Object}   res  Express response
+   * @param  {Function} next Express next handler
+   * @returns {void}
+   */
+  return function middleware(req, res, next) {
+    const header = req.get('authorization');
+    if (header && header.split(' ')[0] === 'Bearer') {
+      req.token = header.split(' ')[1];
```
Comment (melfardi) [inline src/middlewares/jwtToken.js]: We should split header once.

File: test/src/middlewares/childLogger.test.js [hunk 7]
```diff
@@ -0,0 +1,44 @@
+'use strict';
+/* eslint no-unused-expressions:0 */
+
+const expect = require('chai').expect;
+const sinon = require('sinon');
+const childLogger = require('../../../src/middlewares/childLogger.js');
+
+describe('child logger middleware - childLogger.js', function root() {
```
Comment (melfardi) [inline test/src/middlewares/childLogger.test.js]: You can use an arrow function at describe level if you don't use mocha's feature for name and timeout manipulations.

File: test/src/middlewares/childLogger.test.js [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+  it('should throw assertion error when logger is missing', function test() {
+    expect(childLogger).to.throw('AssertionError');
+  });
+
+  it('should work with a requestId', function test(done) {
+    const req = { requestId: '110e8400-e29b-11d4-a716-446655440000' };
+    const logger = { child: () => { } };
```
Comment (melfardi) [inline test/src/middlewares/childLogger.test.js]: You can also use `_.noop` here.

**Comment by melfardi** [comment]
Minor comments, LGTM otherwise.

**Pull request merged**